### PR TITLE
Move AppDomain testing stuff to OrleansTestingHost project

### PR DESCRIPTION
**Pull request by jdom**
This implements #2196 
Move AppDomain testing stuff to OrleansTestingHost project. This is because Orleans will be compiled as .NET Standard, while OrleansTestingHost will continue to be .NET Framework for now (so it can use AppDomains). This will evolve after the CRAWL phase to not rely on AppDomains.
Also abstract it away so that it can be replaced with other strategies in the future (such as processes, containers, etc).

### CODE REVIEW

File: src/OrleansRuntime/Silo/Silo.cs [hunk 1]
```diff
@@ -1054,6 +1038,16 @@ internal void SetMaxForwardCount_ForTesting(int val)
                 silo.OrleansConfig.Globals.MaxForwardCount = val;
             }
 
+            internal void LatchIsOverloaded(bool overloaded)
+            {
+                this.silo.Metrics.LatchIsOverload(overloaded);
```
Comment (sergeybykov) [inline src/OrleansRuntime/Silo/Silo.cs]: >LatchIsOverload [](start = 34, length = 15)

Typo in the method name?

Comment (jdom) [inline src/OrleansRuntime/Silo/Silo.cs]: Yes, but I didn't touch it to avoid unnecessary changes. I can fix if you prefer

Comment (sergeybykov) [inline src/OrleansRuntime/Silo/Silo.cs]: Either way is fine with me.

Comment (jdom) [inline src/OrleansRuntime/Silo/Silo.cs]: K, let's merge as is then. We can decide to fix later if it doesn't bring braking changes

**Pull request merged**